dimod/constrained.py

```diff
@@ -1948,6 +1949,32 @@ def from_lp_file(cls,
 
         return obj
 
+    def __str__(self):
+        vartype_name = {Vartype.SPIN: 'Spin',
+                        Vartype.BINARY: 'Binary',
+                        Vartype.INTEGER: 'Integer',
+                        Vartype.REAL: 'Real'}
+
+        def var_encoder(v):
+            return f'{vartype_name[self.vartype(v)]}({v!r})'
+
+        sio = StringIO()
+        sio.write('Objective\n')
+        sio.write(f'  {self.objective.to_polystring(encoder=var_encoder)}\n')
```

Comment:
not that performance really matters for printing, but this way you're doing a big string copy. I imagine that 
```python
sio.write('  ')
sio.write(self.objective.to_polystring(encoder=var_encoder))
sio.write('\n')
```
will end up being faster for large models.

Reply:
✔️